### ISSUE TITLE
Repeatable directives should not overwrite existing ones

### ISSUE THREAD

**Issue by geshwho** [rmosolgo/graphql-ruby#3987]
**Is your feature request related to a problem? Please describe.**

Given a repeatable directive, if we try to add the same directive to a object twice it will overwrite existing directives of the same class (i.e. last one wins)

**Describe the solution you'd like**

We shouldn't remove existing directives if the class is repeatable. E.g. something like #3986.  

**Describe alternatives you've considered**

None

**Additional context**

Specific context: We're trying to make use of native directives in the [Gusto/apollo-federation-ruby](https://github.com/Gusto/apollo-federation-ruby) library. Apollo federation makes use of repeatable directives (e.g. Key). We need to be able to add multiple keys to a given object.

**Comment by rmosolgo** [comment]
Sorry for the trouble, and thanks for reporting it! The approach you sketched out in #3986 looks promising to me. Are you interested in finishing it out with a test or two?

**Comment by geshwho** [comment]
> Sorry for the trouble, and thanks for reporting it! The approach you sketched out in #3986 looks promising to me. Are you interested in finishing it out with a test or two?

Yep! If the approach makes sense I'll take a stab at finishing it up tonight. Thanks!

**Comment by geshwho** [comment]
@rmosolgo I've added some tests. Let me know what you think!

**Comment by rmosolgo** [comment]
Fixed by #3986 🎉
